### PR TITLE
Verify if output has logits or prediction logits in fill-mask pipeline

### PR DESCRIPTION
**Pull request by pedrogengo**
# What does this PR do?

It checks if the output has the key "logits" or "prediction_logits" to avoid breaking the fill-mask pipeline for some models like BertForPreTraining that returns:

```
return BertForPreTrainingOutput(
            loss=total_loss,
            prediction_logits=prediction_scores,
            seq_relationship_logits=seq_relationship_score,
            hidden_states=outputs.hidden_states,
            attentions=outputs.attentions,
        )
```

Error without this change:

image

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@ArthurZucker @younesbelkada

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Hey! Thanks for this, I am not really against it but most of these models are only there for pretraining, not for inference!

**Comment by pedrogengo** [comment]
Maybe we should have a different error message for these models?
